src/Widgets/PopoverWidget.vala

```diff
@@ -130,4 +144,8 @@ public class Power.Widgets.PopoverWidget : Gtk.Grid {
     private void update_device_separator_revealer () {
         device_separator_revealer.reveal_child = dm.backlight.present && dm.has_battery;
     }
+
+    public void update_power_mode () {
+        power_mode_list?.update_active_profile ();
```

Comment:
Oh this is an interesting syntax I hadn't seen before. Is this a built-in null check? very cool!

Reply:
Yes this is a null-safe expression. This will only run the function if `power_mode_list` is not null. I learnt it in C# and found out Vala supports it too.